giotto/homology/grids.py

```diff
@@ -0,0 +1,182 @@
+    for performance.
+
+    Persistence diagrams produced by this class must be interpreted with
+    care due to the presence of padding triples which carry no information.
+    See :meth:`transform` for additional information.
+
+    References
+    ----------
+    [1] P. Dlotko, "Cubical complex", 2015; `GUDHI User and Reference Manual \
+    <http://gudhi.gforge.inria.fr/doc/latest/group__cubical__complex.html>`_.
+
+    """
+    _hyperparameters = {'infinity_values_': [numbers.Number],
+                        '_homology_dimensions': [list, [int, (0, np.inf)]],
+                        'coeff': [int, (2, np.inf)]}
+
+    def __init__(self, homology_dimensions=(0, 1), coeff=2,
+                 infinity_values=None, n_jobs=None):
+        self.homology_dimensions = homology_dimensions
+        self.coeff = coeff
+        self.infinity_values = infinity_values
+        self.n_jobs = n_jobs
+
+    def _gudhi_diagram(self, X):
+        cubical_complex = CubicalComplex(
+            dimensions=X.shape,
+            top_dimensional_cells=X.flatten(order="F"))
+        Xds = cubical_complex.persistence(homology_coeff_field=self.coeff,
+                                          min_persistence=0)
+        Xds = {dim: np.array([Xds[i][1] for i in range(len(Xds))
+                              if Xds[i][0] == dim]).reshape((-1, 2))
+               for dim in self.homology_dimensions}
+        Xds = {dim: np.hstack([Xds[dim], dim * np.ones((Xds[dim].shape[0], 1),
+                                                       dtype=Xds[dim].dtype)])
+               for dim in self._homology_dimensions}
+        return Xds
+
+    def fit(self, X, y=None):
+        """Do nothing and return the estimator unchanged.
+
+        This method is there to implement the usual scikit-learn API and hence
+        work in pipelines.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_samples, n_pixels_1, ..., n_pixels_d)
+            Input data. Array of d-dimensional images.
+
+        y : None
+            There is no need of a target in a transformer, yet the pipeline API
+            requires this parameter.
+
+        Returns
+        -------
+        self : object
+
+        """
+        if self.infinity_values is None:
+            self.infinity_values_ = np.max(X)
+        else:
+            self.infinity_values_ = self.infinity_values
+
+        self._homology_dimensions = sorted(self.homology_dimensions)
+
+        validate_params({**self.get_params(),
+                         'infinity_values_': self.infinity_values_,
+                         '_homology_dimensions': self._homology_dimensions},
+                        self._hyperparameters)
+        check_array(X, allow_nd=True)
+
+        self._max_homology_dimension = self._homology_dimensions[-1]
+        return self
+
+    def transform(self, X, y=None):
+        """Compute, for each image in `X`, the relevant persistence diagram
```

Comment:
I would write this as "For each image in `X`, compute the corresponding persistence diagram ..."

Reply:
It was written mimicking [VietorisRipsPersisentence](https://github.com/giotto-ai/giotto-learn/blob/master/giotto/homology/point_clouds.py#L160) but I dont' mind making the change.